activation/handler_v1_test.go

```diff
@@ -655,6 +721,147 @@ func TestHandlerV1_StoreAtx(t *testing.T) {
+		atx1 := newChainedActivationTxV1(t, initialATX, goldenATXID)
+		atx1.Sign(sig)
+		atxHdlr.mValidator.EXPECT().IsVerifyingFullPost()
+		atxHdlr.mbeacon.EXPECT().OnAtx(gomock.Cond(func(atx any) bool {
+			return atx.(*types.ActivationTx).ID() == atx1.ID()
+		}))
+		atxHdlr.mtortoise.EXPECT().OnAtx(atx1.PublishEpoch+1, atx1.ID(), gomock.Any())
+		_, proof, err = atxHdlr.storeAtx(
+			context.Background(),
+			atx1, nil,
+			time.Now(), atxHdlr.tickSize, atx1.NumUnits,
+		)
+		require.NoError(t, err)
+		require.Nil(t, proof)
+
+		atx2 := newChainedActivationTxV1(t, atx1, goldenATXID)
+		atx2.Sign(sig)
+		atxHdlr.mValidator.EXPECT().IsVerifyingFullPost()
+		atxHdlr.mbeacon.EXPECT().OnAtx(gomock.Cond(func(atx any) bool {
+			return atx.(*types.ActivationTx).ID() == atx2.ID()
+		}))
+		atxHdlr.mtortoise.EXPECT().OnAtx(atx2.PublishEpoch+1, atx2.ID(), gomock.Any())
+		_, proof, err = atxHdlr.storeAtx(
+			context.Background(),
+			atx2, nil,
+			time.Now(), atxHdlr.tickSize, atx2.NumUnits,
+		)
+		require.NoError(t, err)
+		require.Nil(t, proof)
+
+		// third non-initial ATX references initial ATX as prevATX
+		atx3 := newChainedActivationTxV1(t, initialATX, goldenATXID)
+		atx3.PublishEpoch = atx2.PublishEpoch + 1
+		atx3.Sign(sig)
+		atxHdlr.mValidator.EXPECT().IsVerifyingFullPost()
+		atxHdlr.mbeacon.EXPECT().OnAtx(gomock.Cond(func(atx any) bool {
+			return atx.(*types.ActivationTx).ID() == atx3.ID()
+		}))
+		atxHdlr.mtortoise.EXPECT().OnAtx(atx3.PublishEpoch+1, atx3.ID(), gomock.Any())
+		atxHdlr.mtortoise.EXPECT().OnMalfeasance(sig.NodeID())
+		atxHdlr.mclock.EXPECT().CurrentLayer().Return(atx3.PublishEpoch.FirstLayer())
+		_, proof, err = atxHdlr.storeAtx(
+			context.Background(),
+			atx3, nil,
+			time.Now(), atxHdlr.tickSize, atx3.NumUnits,
+		)
+		require.NoError(t, err)
+		require.NotNil(t, proof)
+		require.Equal(t, mwire.InvalidPrevATX, proof.Proof.Type)
+		proof.SetReceived(time.Time{})
+		nodeID, err := malfeasance.Validate(
+			context.Background(),
+			atxHdlr.log,
+			atxHdlr.cdb,
+			atxHdlr.edVerifier,
+			nil,
+			&mwire.MalfeasanceGossip{
+				MalfeasanceProof: *proof,
+			},
+		)
+		require.NoError(t, err)
+		require.Equal(t, sig.NodeID(), nodeID)
+	})
+
+	t.Run("another atx with the same prevatx for registered ID doesn't create a malfeasance proof", func(t *testing.T) {
+		atxHdlr := newV1TestHandler(t, goldenATXID)
+		atxHdlr.Register(sig)
+
+		// Act & Assert
+		initialATX := newInitialATXv1(t, goldenATXID)
+		initialATX.Sign(sig)
+
+		atxHdlr.mValidator.EXPECT().IsVerifyingFullPost()
+		atxHdlr.mbeacon.EXPECT().OnAtx(gomock.Cond(func(atx any) bool {
+			return atx.(*types.ActivationTx).ID() == initialATX.ID()
+		}))
+		atxHdlr.mtortoise.EXPECT().OnAtx(initialATX.PublishEpoch+1, initialATX.ID(), gomock.Any())
+		_, proof, err := atxHdlr.storeAtx(
+			context.Background(),
+			initialATX, nil,
+			time.Now(), atxHdlr.tickSize, initialATX.NumUnits,
+		)
+		require.NoError(t, err)
+		require.Nil(t, proof)
```

Comment:
How about pushing the ATX though the handler instead? The test would be more concise this way:
```suggestion
		atxHdlr.atxHdlr.expectAtxV1(initialATX, sig.NodeID()))
		_, proof, err := := atxHdlr.processATX(context.Background(), "", initialATX, codec.MustEncode(initialATX), time.Now())
		require.NoError(t, err)
		require.Nil(t, proof)
```

Reply:
I'm testing the `storeAtx` function specifically, not the whole `processATX` which does much more?

Reply:
The part I suggest changing is not testing `storeAtx() method, it's preparing for testing it below. I recommend simplifying the test so it's less cluttered.